app/lib/ldap_authenticatable_tiny/strategy.rb

```diff
@@ -29,14 +32,14 @@ def authenticate!
       private
       # Create a Net::LDAP instance, `bind` to it and close.
       # Return true or false if they're authorized.
-      def is_authorized_by_ldap?(login, password)
+      def is_authorized_by_ldap?(ldap_service, login, password)
         return false if password.nil? || password == ''
         options = ldap_options_for(login, password)
-        ldap = Net::LDAP.new(options)
+        ldap = ldap_service.new(options)
         is_authorized = ldap.bind
 
         if !is_authorized
-          ldap_error = ldap.get_operation_result
+          ldap_error = ldap_service.get_operation_result
```

Comment:
Will this have to change?  I was thinking our dummy class would return an instance that implemented this method so this would stay `ldap.get_operation_result`.

Reply:
Oh good catch here.